### ISSUE TITLE
How to create pset with issuance

### ISSUE THREAD

**Issue by altafan** [ElementsProject/elements#1091]
Hello,

I'm trying to use the CLI to create a pset (v2 format) in order to issue a new asset. For simplicity, I'm going for a fully unblinded pset. These are my steps:

```
$ e1 sendtoaddress ${issuer_unconf_addr_1} 1
$ e1 generatetoaddress 1 $(e1 getnewaddress)

# fund the issuer and retrie ${issuer_utxp_txid} and ${issuer_utxo_vout}

$ e1 calculateasset ${issuer_utxo_txid} ${issuer_utxo_vout} 0000000000000000000000000000000000000000000000000000000000000000 false

# retrieve ${issued_asset} and ${issued_token}

$ e1 walletcreatefundedpsbt '[{"txid": "${issuer_utxo_txid}", "vout": ${issuer_utxo_vout}, "asset_entropy": "0000000000000000000000000000000000000000000000000000000000000000", "issuance_amount": 1000, "issuance_tokens": 1, "blind_reissuance": false}]' '[{"${issuer_unconf_addr_1}": 0.00001, "asset": "${issued_asset}"}, {"${issuer_unconf_addr_2}": 0.00000001, "asset": "${issued_token}"}]'
```

This always produces the error:

```
error code: -4
error message:
Insufficient funds
```

Can someone help me figuring out what's wrong with this?

**Comment by apoelstra** [comment]
Thanks for filing this.

What version of Elements are you using? I notice that you tagged this `GUI` but the issue is about CLI commands -- can you confirm this this only a problem with the CLI?

**Comment by altafan** [comment]
Yeah, sorry about that, i didn't notice the label was already there when I opened the issue.. this is relative to the CLI, not related to GUI.

I'm running version: `Elements Core RPC client version elements-0.21.0.1`

**Comment by k-matsuzawa** [comment]
I tried the PSBT commands (converttopsbt, walletprocesspsbt, finalizepsbt) and got the same result (problem). The problem seems to be with finalizepsbt.

https://github.com/ElementsProject/elements/blob/elements-0.21.0.1/src/psbt.cpp#L126-L132

- It would be better to match the following implementation of txout.
  - https://github.com/ElementsProject/elements/blob/elements-0.21.0.1/src/psbt.cpp#L140-L152
- It seems that `txin.assetIssuance.nAmount` and `txin.assetIssuance.nInflationKeys` should be judged separately. Because in the case of reissuance, only `txin.assetIssuance.nAmount` is used.
